File: HW/HW6/HW6Bruteforce.py
def threeSum(nums):
    result = []
    n = len(nums)

    for i in range(n - 2):
        for j in range(i + 1, n - 1):
            for k in range(j + 1, n):
                if nums[i] + nums[j] + nums[k] == 0:
                    triplet = [nums[i], nums[j], nums[k]]
                    if triplet not in result:  # Check for duplicates
                        result.append(triplet)
    return result

File: HW/HW6/HW6Bruteforce.java
public List<List<Integer>> threeSum(int[] nums) {
    List<List<Integer>> result = new ArrayList<>();
    int n = nums.length;

    for (int i = 0; i < n - 2; i++) {
        for (int j = i + 1; j < n - 1; j++) {
            for (int k = j + 1; k < n; k++) {
                if (nums[i] + nums[j] + nums[k] == 0) {
                    List<Integer> triplet = Arrays.asList(nums[i], nums[j], nums[k]);
                    if (!result.contains(triplet)) { // Check for duplicates
                        result.add(triplet);
                    }
                }
            }
        }
    }
    return result;
}
